esphome/components/espnow/espnow.cpp

```diff
@@ -0,0 +1,549 @@
+#include "esphome/core/application.h"
+#include "esphome/core/version.h"
+#include "esphome/core/log.h"
+
+#if defined(USE_ESP32)
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 1)
+
+#include "espnow.h"
+
+#include <cstring>
+
+#include "esp_mac.h"
+#include "esp_random.h"
+
+#include "esp_wifi.h"
+#include "esp_now.h"
+#include "esp_event.h"
+
+#ifdef USE_WIFI
+#include "esphome/components/wifi/wifi_component.h"
+#endif
+
+#include <memory>
+
+namespace esphome {
+namespace espnow {
+
+static const char *const TAG = "espnow";
+const char *const ESPNowTAG::TAG = "espnow";
```

Comment:
```suggestion
static constexpr auto TAG = ESPNowTAG::TAG;
```
This should work, ESPNowTAG::TAG is just cosnt pointer to const char, TAG is also the same. Should save s bit of memory by not storing 2 string literals and 2 pointers in ram